tools/infrasys/src/main.rs

```diff
@@ -0,0 +1,767 @@
+        create_keys(signing_keys).await; // TODO: add an ensure! ok check (make sure RESULT is used)
+        create_keys(root_keys).await;
+
+        // Step 4: Create and sign root.json
+        tuftool!(Region::default().name(), "root init '{}'", outdir.display());
+        tuftool!(
+            Region::default().name(),
+            // IN-FUTURE: expose expiration date as a configurable parameter
+            "root expire '{}' 'in 52 weeks'",
+            outdir.display()
+        );
+        // Root signing
+        sign_root(
+            get_available_keys(signing_keys)?, // QUESTION: What is the workflow of errors percolating up? Do I have to do any explicit error catching if the method()? resolves to error?
+            &KeyRole::Publication,
+            repo_config
+                .pub_key_threshold
+                .as_ref()
+                .context(error::MissingConfig {
+                    missing: format!("pub_key_threshold for '{}' repo config", repo_name),
+                })?,
+            &outdir.display().to_string(),
+        );
+        // TODO test root signing
+        sign_root(
+            get_available_keys(root_keys)?,
+            &KeyRole::Root,
+            repo_config
+                .root_key_threshold
+                .as_ref()
+                .context(error::MissingConfig {
+                    missing: format!("root_key_threshold for '{}' repo config", repo_name),
+                })?,
+            &outdir.display().to_string(),
+        );
+
+        // Step 5: Upload root.json
+        upload_file(s3_region, bucket_name, prefix, outdir).await;
+
+        // Step 6: Update output paramters
+        // TODO: don't override if urls are not none 
+        // - repo_config.metadata_base_url = Some(Url::parse(format!("{}{}/metadata/", bucket_url, prefix)));
+        // - repo_config.targets_url = Some(format!("{}{}/targets/", bucket_url, prefix));
+        // - repo_config.root_role_url = Some(format!("{}{}/root.json", bucket_url, prefix));
+        // https://docs.rs/sha2/0.9.5/sha2/
+    }
+
+    // Step 7: Generate Infra.lock
+    // TODO: create infra.lock
+
+    // - let toml_string = toml::to_string(&config).unwrap();
+    // - fs::write("Infra.lock", toml_string).expect("Could not write to file!");
+    // - println!("{:#?}", infra_config);
+
+    Ok(())
+}
+
+//  =^..^=   =^..^=   =^..^= S3 BUCKET FUNCTIONALITY  =^..^=   =^..^=   =^..^=
+
+/// Creates a *private* S3 Bucket using a CloudFormation template
+/// Input: The region in which the bucket will be created and the name of the bucket
+/// Output: The stack_arn of the bucket created (will be added as a field to Infra.lock)
+async fn create_s3_bucket(region: &String, bucket_name: &String) -> Result<String> {
+    // TO DO: consider what to do if bucket already exists? We are supposed to support that -- will get "bucket already exists" error
+    let stack_name = String::from("TUF-Repo"); // QUESTION: how to avoid these "string froms"?
```

Comment:
`String::from` to create String is correct. You can do `"TUF_Repo".to_string()` if you prefer it that way. But you cannot avoid the conversion from str slice to String.

Reply:
Also, `TUF-Repo` is too generic for the stack name. It should be some combination of "tuf_infra" and "s3" and the s3 key label from Infra.toml